v2/devdoc/log_context_property_type_wchar_t_ptr_requirements.md

```diff
@@ -0,0 +1,95 @@
+# `log_context_property_type_wchar_t_ptr` requirements
+
+`log_context_property_type_wchar_t_ptr` implements the code needed for handling an wchar_t string.
+
+## Exposed API
+
+```
+int LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(void* dst_value, size_t count, const wchar_t* format, ...);
+int LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(const wchar_t* format, ...);
+
+extern const LOG_CONTEXT_PROPERTY_TYPE_IF LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr);
+
+```
+
+## LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string
+
+```c
+typedef int (*LOG_CONTEXT_PROPERTY_TYPE_TO_STRING)(const void* property_value, char* buffer, size_t buffer_length);
+```
+
+`LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` produces the string representation of a `wchar_t_ptr`.
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_001: [** If `property_value` is `NULL`, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall fail and return a negative value.  **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_002: [** If `buffer` is `NULL` and `buffer_length` is greater than 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall fail and return a negative value. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_003: [** If `buffer` is `NULL` and `buffer_length` is 0, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall return the length of the `wchar_t` string pointed to by `property_value`. **]**
+
+**SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_004: [** Otherwise, `LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).to_string` shall copy the wchar_t string pointed to by `property_value` to `buffer` by using `snprintf` with `buffer`, `buffer_length` and format string `%ls` and pass in the values list the `const wchar_t*` value pointed to be `property_value`. **]**
```

Comment:
snprintf doesn't work for linux, modified to use wcstombs here. 

ascii_char_ptr_to_string also use snprintf, the copying of the string can be somewhat optimized by not using `snprintf`, this will be considered in a separate task.